kb/all-the-ways-to-run-a-cluster.md

```diff
@@ -0,0 +1,123 @@
+
+The main reason you might want to do this is to pass in custom plugins or any other customizations through the [Server Options](/references/server-options).
+Then you can build and run a binary that contains your customizations.
+
+Doing this requires [Go v1.16+](https://github.com/temporalio/temporal/blob/master/CONTRIBUTING.md).
+
+## Temporal Server as a binary
+
+You can run the Temporal Server as a single Go binary, or you can run each service within the Server separately.
+For example, if you are using Kubernetes, you could have one service per pod so they can be scaled independently in future.
+
+In Docker, you could run each service in its own container, using the `SERVICES` flag to specify the service:
+
+```bash
+docker run
+    # persistence/schema setup flags omitted
+    -e SERVICES=history \                      -- Spin up one or more: history, matching, worker, frontend
+    -e LOG_LEVEL=debug,info \                           -- Logging level
+    -e DYNAMIC_CONFIG_FILE_PATH=config/foo.yaml         -- Dynamic config file to be watched
+    temporalio/server:<tag>
+```
+
+For more details, see the [Docker source file](https://github.com/temporalio/temporal/tree/master/docker).
+
+Each Temporal Server release also ships a `Server with Auto Setup` Docker image that includes an [`auto-setup.sh` script](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) we recommend using for initial schema setup of each supported database.
+You should familiarize yourself with [what auto-setup does](https://temporal.io/blog/auto-setup), because you will likely replace every part of the script to customize it for your own infrastructure and tooling choices.
+
+## Gitpod
+
+You can run a Temporal Cluster, run Worker processes, and develop Temporal Applications in your browser using [Gitpod](https://www.gitpod.io/).
+
+One-click deployments are available for the [temporalio/samples-go](https://github.com/temporalio/samples-go) repo and the [temporalio/samples-typescript](https://github.com/temporalio/samples-typescript) repo.
+This approach runs a Cluster using a Temporal Server Docker image, starts a Worker Process, and starts one of the application's sample Workflows.
+
+A one-click deployment can take up to a full minute to get fully up and running.
+When it is running, you can customize the application samples.
+This approach does not offer features such as Advanced Visibility or Archival.
+
+This approach is often used for ephemeral purposes, such as learning and demos.
+
+## Helm charts
+
+[Temporal Helm charts](https://github.com/temporalio/helm-charts) enables you to get a Cluster running on [Kubernetes](https://kubernetes.io/) by deploying the Temporal Server services to individual pods and connecting them to your existing database and Elasticsearch instances.
+
+This approach requires a lot of customizations and can easily become very complex if you try to scale services or run many Workflows concurrently.
```

Comment:
The template in temporalio/helm-charts repo is your starting point, but you should adjust to fit your infrastructure needs, it is highly customizable.